CRM/Contribute/BAO/Contribution.php

```diff
@@ -539,6 +540,22 @@ public static function create(&$params) {
         $activityParams['source_contact_id'] = (int) ($params['source_contact_id'] ?? (CRM_Core_Session::getLoggedInContactID() ?: $contribution->contact_id));
         $activityParams['target_contact_id'] = ($activityParams['source_contact_id'] === (int) $contribution->contact_id) ? [] : [$contribution->contact_id];
       }
+      else {
+        // Check if target contact exists
```

Comment:
The comments on lines 537-539 should be removed if we're now setting target contacts on update.